sockeye/training.py

```diff
@@ -280,6 +281,7 @@ def _fit(self,
         :param mxmonitor: Optional MXNet monitor instance.
         """
         metric_train = self._create_eval_metric(metrics)
+        metric_batch = self._create_eval_metric(metrics)
```

Comment:
General question about what to pass to the extended optimizers: we mostly track perplexity on training and validation data sets but actually optimize cross-entropy. While strongly related, have you tried passing cross-entropy values to Eve instead? Also, since using SoftmaxOutput gradient already entails cross-entropy loss, we could directly use the CE metric here and would not need to create other metrics (e.g. Accuracy). This would also allow you to get rid of these repeated for loops over the metrics below, by specifically creating an 'eve metric'.

We should also make sure that this won't interact strangely with any sequence losses from the MRT branch, or maybe more importantly, how we normalize/scale the loss. This will affect the size of the gradients you get in Eve.update() but won't be reflected in the batch metric here.

Reply:
Right now Eve uses whatever we specify as `optimized_metric`.  If the optimizer is making updates based on a different loss function, we should be able to use that as well.  Do we know if/where batch loss is stored in a form we can access?

Reply:
Just to make sure this is clear: optimized_metric does not define the actual loss function that is optimized by MXNet (this refers to the metric that is used to compute when to do early stopping w.r.t. validation scores). That said, we currently always minimize cross-entropy on the training set as given by the symbol created in loss.py. If you use smoothed-cross-entropy you should get a grouped symbol of logits and the loss, whereas for the standard cross-entropy loss we use the SoftmaxOutput symbol which returns logits in the forward, and backprobs cross-entropy gradients, so you don't have direct access to the loss function value for a batch.

So what I would suggest is to create a cross-entropy metric (``mx.metric.create('ce')``) to be used with the Eve optimizer so that it gets the correct loss function values for scaling.